website/public/js/services/questServices.js

```diff
@@ -106,20 +97,20 @@
 
     function initQuest(key) {
       return $q(function(resolve, reject) {
-        Analytics.track({'hitType':'event','eventCategory':'behavior','eventAction':'quest','owner':true,'response':'accept','questName': key});
-        Analytics.updateUser({'partyID':party._id,'partySize':party.memberCount});
-        party.$startQuest({key:key}, function(){
-          party.$syncParty();
-          $state.go('options.social.party');
-          resolve();
-        });
+        Analytics.track({'hitType':'event', 'eventCategory':'behavior', 'eventAction':'quest', 'owner':true, 'response':'accept', 'questName': key});
+        Analytics.updateUser({'partyID': party._id, 'partySize': party.memberCount});
+        Groups.Group.startQuest(party._id, key)
+          .then(function() {
+            Groups.Group.syncParty();
```

Comment:
instead of syncing, can't you use the return data from this route? Most quest routes return party.quest so you can just substitute the cached party.quest object with the one returned by the route call.